src/audio/volume/volume.c

```diff
@@ -1217,6 +1215,39 @@ static vol_zc_func vol_get_zc_function(struct comp_dev *dev)
 	return NULL;
 }
 
+/**
+ * \brief Set volume frames alignment limit.
+ * \param[in,out] source Source structure.
+ * \param[in,out] sink Sink structure.
+ */
+static void volume_set_alignment(struct audio_stream *source,
+				 struct audio_stream *sink)
+{
+#if XCHAL_HAVE_HIFI3 || XCHAL_HAVE_HIFI4
+
+	/* both source and sink buffer in HiFi 3 or HiFi4 processing version,
+	 * xtensa intrinsics ask for 8-byte aligned. 5.1 format SSE audio
+	 * requires 16-byte aligned.
+	 */
+	const uint32_t byte_align = source->channels == 6 ? 16 : 8;
```

Comment:
Just curious the logic for having this in volume rather than component. This appears to be generic to the ISA, not the processing functions.

Reply:
@andrula-song please don't close a comment that is asking a question without answering it, especially when you made no code change to indicate resolution.